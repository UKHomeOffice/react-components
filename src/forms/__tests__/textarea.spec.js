import React from 'react'
import { render, shallow } from 'enzyme'
import TextArea from 'govuk-react-components/forms/textarea';

describe('The TextArea Component', () => {

  it('should render the value as content when a value is passed in', () => {
    expect(
      render(<TextArea label="ip" name="ip" value="someText" />)
    ).toMatchSnapshot()
  });

  it('should render a hint when a hint is passed in', () => {
    expect(
      render(<TextArea label="ip" name="ip" value="someText" hint="Testing hint" />)
    ).toMatchSnapshot();
  });

  it('should render an error when an error is passed in', () => {
    expect(
      render(<TextArea label="ip" name="ip" value="someText" error="Testing error" />)
    ).toMatchSnapshot();
  });

  it('should call the onChange function when a change is simulated', () => {
    const newVal = 'someText';
    const onChange = jest.fn();
    const field = shallow(<TextArea label="ip" name="ip" onChange={onChange} />);
    field.find('textarea').simulate('change', newVal);
    expect(onChange).toBeCalledWith(newVal);
  });

  it('should render cols & rows when props are given', () => {
    expect(
      render(<TextArea label="ip" name="ip" value="someText" rows={8} cols={10}/>)
    ).toMatchSnapshot()
  })
})
