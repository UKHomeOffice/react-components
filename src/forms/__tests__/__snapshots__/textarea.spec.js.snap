// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Input Text Component should render a hint when a hint is passed in 1`] = `
<div
  class="form-group"
>
  <label
    class="form-label"
    for="ip"
  >
    <span
      class="form-label-bold"
    >
      ip
    </span>
    <span
      class="form-hint"
    >
      Testing hint
    </span>
  </label>
  <textarea
    class="form-control"
    id="ip"
    name="ip"
    rows="4"
  >
    someText
  </textarea>
</div>
`;

exports[`The Input Text Component should render an error when an error is passed in 1`] = `
<div
  class="form-group form-group-error"
>
  <label
    class="form-label"
    for="ip"
  >
    <span
      class="form-label-bold"
    >
      ip
    </span>
    <span
      class="error-message"
    >
      Testing error
    </span>
  </label>
  <textarea
    class="form-control form-control-error"
    id="ip"
    name="ip"
    rows="4"
  >
    someText
  </textarea>
</div>
`;

exports[`The Input Text Component should render as a checked input correctly 1`] = `
<div
  class="form-group"
>
  <label
    class="form-label"
    for="ip"
  >
    <span
      class="form-label-bold"
    >
      ip
    </span>
  </label>
  <textarea
    class="form-control"
    id="ip"
    name="ip"
    rows="4"
  >
    someText
  </textarea>
</div>
`;

exports[`The Input Text Component should render cols & rows 1`] = `
<div
  class="form-group"
>
  <label
    class="form-label"
    for="ip"
  >
    <span
      class="form-label-bold"
    >
      ip
    </span>
  </label>
  <textarea
    class="form-control"
    cols="10"
    id="ip"
    name="ip"
    rows="8"
  >
    someText
  </textarea>
</div>
`;

exports[`The Input Text Component should render correctly when a value is passed in 1`] = `
<div
  class="form-group"
>
  <label
    class="form-label"
    for="ip"
  >
    <span
      class="form-label-bold"
    >
      ip
    </span>
  </label>
  <textarea
    class="form-control"
    id="ip"
    name="ip"
    rows="4"
  >
    someText
  </textarea>
</div>
`;
